src/network/teldat/snmp/mode/cellsradio.pm

```diff
@@ -0,0 +1,346 @@
+};
+
+my $oid_teldatCellularInfoInterfaceEntry = '.1.3.6.1.4.1.2007.4.1.2.2.2.18.1.1'; # teldatInfoInterfaceTable
+my $oid_teldatCellularStateMobileEntry = '.1.3.6.1.4.1.2007.4.1.2.2.2.18.3.2.1'; # teldatStateMobileTable
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    $self->{global} = { detected => 0 };
+    $self->{cells} = {};
+
+    my $snmp_result = $options{snmp}->get_multiple_table(
+        oids => [
+            { oid => $mapping_state_interface->{interfaceState}->{oid} },
+            { oid => $mapping_prof_dial->{operator}->{oid} },
+            { oid => $oid_teldatCellularInfoInterfaceEntry, start => $mapping_info_interface->{imei}->{oid}, end => $mapping_info_interface->{simIcc}->{oid} },
+            { oid => $oid_teldatCellularStateMobileEntry, start => $mapping_state_mobile->{techno}->{oid}, end => $mapping_state_mobile->{simStatus}->{oid} }
+        ],
+        nothing_quit => 1
+    );
+
+    foreach my $oid (keys %{$snmp_result->{$oid_teldatCellularInfoInterfaceEntry}}) {
+        next if ($oid !~ /^$mapping_info_interface->{imei}->{oid}\.(.*)$/);
+        my $instance = $1;
+
+        my $result = $options{snmp}->map_instance(mapping => $mapping_info_interface, results => $snmp_result->{$oid_teldatCellularInfoInterfaceEntry}, instance => $instance);
+        my $result2 = $options{snmp}->map_instance(mapping => $mapping_state_interface, results => $snmp_result->{$mapping_state_interface->{interfaceState}->{oid}}, instance => $instance);
+        my $result3 = $options{snmp}->map_instance(mapping => $mapping_prof_dial, results => $snmp_result->{$mapping_prof_dial->{operator}->{oid}}, instance => $instance);
+
+        my $cell_id = $result->{imei};
+        next if ($cell_id !~ /^(?:[0-9]+)$/);
+        next if (defined($self->{option_results}->{filter_cell_id}) && $self->{option_results}->{filter_cell_id} ne '' &&
+            $cell_id !~ /$self->{option_results}->{filter_cell_id}/ && $result->{simIcc} !~ /$self->{option_results}->{filter_cell_id}/);
+
+        my $operator = $result3->{operator};
+        if($result3->{operator} =~ /^-+$/){
+            $operator = "N/A";
+        }
+        $self->{cells}->{$instance} = {
+            cellId => $cell_id,
+            simIcc  => $result->{simIcc},
+            operator => $operator,
+            status => {
+                cellId => $cell_id,
+                simIcc  => $result->{simIcc},
+                operator => $operator,
+                imsi   => $result->{imsi},
+                interfaceState => $result2->{interfaceState}
+            },
+            signal => {
+                cellId => $cell_id,
+                simIcc  => $result->{simIcc},
+                operator => $operator
+            }
+        };
+    }
+
+    if (scalar(keys %{$self->{cells}}) <= 0 && defined($self->{option_results}->{filter_cell_id}) && $self->{option_results}->{filter_cell_id} ne '') {
```

Comment:
@omercier Avoid OK: Empty output if the filter didn't catch anything (see below). But maybe in link with the overall issue with OK: Empty outputs.

```
# Return : OK:  | 'modules.cellradio.detected.count'=0;;;0;
    # return if (scalar(keys %{$self->{cells}}) <= 0);
```